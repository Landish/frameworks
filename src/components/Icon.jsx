import React from 'react';

const Icon = ({ name }) => {
  return (
    <span className="icon is-small">
      <i className={`fa fa-${name}`}/>
    </span>
  );
};

export default Icon;