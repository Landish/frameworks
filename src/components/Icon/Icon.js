import React, { PureComponent } from 'react';
import { string } from 'prop-types';

class Icon extends PureComponent {
  static propTypes = {
    name: string.isRequired
  };

  render() {
    return (
      <span className="icon is-small">
        <i className={`fa fa-${this.props.name}`} />
      </span>
    );
  }
}

export default Icon;
