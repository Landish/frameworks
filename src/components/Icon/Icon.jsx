import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ name }) => {
  return (
    <span className="icon is-small">
      <i className={`fa fa-${name}`} />
    </span>
  );
};

Icon.propTypes = {
  name: PropTypes.string.isRequired
};

export default Icon;
