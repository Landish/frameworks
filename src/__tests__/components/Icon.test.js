import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../components/Icon';

it('renders correctly', () => {
  const tree = renderer.create(<Icon name="github" />).toJSON();
  expect(tree).toMatchSnapshot();
});
