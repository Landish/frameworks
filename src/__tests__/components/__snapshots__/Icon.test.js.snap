// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="icon is-small"
>
  <i
    className="fa fa-github"
  />
</span>
`;
